code/datums/proximity_monitor/field.dm

```diff
@@ -64,16 +76,24 @@
 	else
 		field_turf_uncrossed(gone, source)
 
+/// Called when a turf in the field of the monitor is linked
 /datum/proximity_monitor/advanced/proc/setup_field_turf(turf/target)
-	field_turfs |= target
+	return
 
+/// Called when a turf in the field of the monitor is unlinked
 /datum/proximity_monitor/advanced/proc/cleanup_field_turf(turf/target)
 	field_turfs -= target
 
+/// Called when a turf in the edge of the monitor is linked
 /datum/proximity_monitor/advanced/proc/setup_edge_turf(turf/target)
-	edge_turfs |= target
+	if(edge_is_a_field) // If the edge is considered a field, set it up like one
+		setup_field_turf(target)
+	return
```

Comment:
rewrite it so we early return if `edge_is_a_field` is false, or at least remove the redundant return at the end of the proc